openslides_backend/action/actions/user/toggle_presence_by_number.py

```diff
@@ -60,21 +60,23 @@ def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
         return instance
 
     def find_user_to_number(self, meeting_id: int, number: str) -> int:
-        filter_: Filter = FilterOperator(f"number_${meeting_id}", "=", number)
-        result = self.datastore.filter("user", filter_, ["id"])
+        filter_: Filter = FilterOperator("number", "=", number)
+        result = self.datastore.filter("meeting_user", filter_, ["user_id"])
         if len(result.keys()) == 1:
-            return list(result.keys())[0]
+            return list(result.values())[0]["user_id"]
         elif len(result.keys()) > 1:
             raise ActionException("Found more than one user with the number.")
 
-        filter_ = And(
-            FilterOperator(f"number_${meeting_id}", "=", ""),
-            FilterOperator("default_number", "=", number),
-        )
+        filter_ = FilterOperator("number", "=", "")
```

Comment:
Maybe it's easier to look for the default_number in the user-collection.
On 0 results, there is no user with this number, on 1 or more results read meeting_user with a filter with meeting_id and empty number. 
The intersection-set of both resulting user_ids has to have 1 element or is an error.

Reply:
Start with looking for the number in the user/default_numbers, and then filter for empty string in number and meeting_id.